nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-jetty/src/main/java/org/apache/nifi/web/server/JettyServer.java

```diff
@@ -724,7 +726,7 @@ protected List<URI> getApplicationUrls() {
                     final String scheme = sslConnectionFactory == null ? HTTP_SCHEME : HTTPS_SCHEME;
                     final int port = serverConnector.getLocalPort();
                     final String connectorHost = serverConnector.getHost();
-                    final String host = StringUtils.defaultIfEmpty(connectorHost, HOST_UNSPECIFIED);
+                    final String host = URLEncoder.encode(StringUtils.defaultIfEmpty(connectorHost, HOST_UNSPECIFIED), StandardCharsets.UTF_8);
                     return URI.create(String.format(APPLICATION_URL_FORMAT, scheme, host, port));
```

Comment:
Instead of using `URLEncoder` would using the seven argument constructor of `java.net.URI` be able to resolve to a URL?
I was able to take the parts of the URL you listed in the ticket and instantiate the following in a unit test
`URI uri = new URI("https", null, "0:0:0:0:0:0:0:1%lo", 8443, "/nifi", null, null);`

I am just unsure how `%lo` gets resolved.

Reply:
Thanks @dan-s1 , that seems like a simpler solution and does appear to work, the logs now show:
```
2024-02-20 12:06:40,195 INFO [main] org.apache.nifi.web.server.JettyServer Started Server on https://[fe80:0:0:0:c6d:9e76:6aff:9d0f%en0]:8443/nifi
2024-02-20 12:06:40,195 INFO [main] org.apache.nifi.web.server.JettyServer Started Server on https://192.168.1.169:8443/nifi
```